### ISSUE TITLE
Changes to games / profiles not saving / regularly disappear

### ISSUE THREAD

**Issue by PoliteTimesplitter** [x360ce/x360ce#1484]
I'm trying to set up x360CE so it only activates when playing two games (YARG and Guitar Hero World Tour Definitive Edition), and I'm putting my xbox 360 guitar and USB foot pedal (which is itself set to a keyboard input) through x360CE. Both the 'back' button on the guitar and the foot pedal are set to 'back' on the emulated controller. I have added two games - the respective EXEs - and I'll set up the buttons, hit 'Save All' and then, when I flick to another one in the dropdown menu (or the x360ce.exe game listing itself), and then back again, the inputs have just disappeared.

I get a 'Errors: 2' message whenever I open x360CE as well, and this is what follows:

> System.InvalidOperationException: Selected property cannot be altered before this object is added to a DataGridView control.
> 
> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
> 
> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
> 
> System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.

**Comment by EvoJocys** [comment]
This looks like it was addressed. From the changelog:

4.17.12 — Fixed: Issue when mappings from other games would merge into currently selected game.
4.18.63 — Fixed: Cancelling the add device window added the devices chosen the time before.
4.21.30 — Fixed: The games list could close the program while a game was added or removed.

Please try 4.21: https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
